### DISCUSSION TITLE
Compatibility question

### DISCUSSION THREAD

**Discussion by davidgrohl** [h2zero/NimBLE-Arduino#370] [Q&A]
Hi all,
for a new DIY project I'm selecting an ESP32 device. I would like to know if this library is compatible with both ESP32-WROOM and the new ESP32-S3 device. 
Thanks in advance.

**Comment by h2zero** [comment]
Hello @davidgrohl, yes this library works with any BLE capable ESP32 device.

**Reply by davidgrohl** [reply]
Thanks @h2zero! I would like to create an HID device to connect to raspberry pi. Do you think it can work? I mean, do you think esp32 will pair with raspberry?

**Reply by h2zero** [reply]
Yes that will work just fine, all the tools needed are available here 😄

**Reply by davidgrohl** [reply]
Thank you so much. I’ll come back to get some help in next weeks 😅

**Reply by h2zero** [reply]
Sure thing 😃

**Reply by davidgrohl** [reply]
I retrieved a ESP32 moduel. It's a ESP32-S2-Saola-1: can I use it with this library?

**Reply by h2zero** [reply]
The esp32-s2 does not have Bluetooth capability, so that module will not be able to use this library.

**Reply by davidgrohl** [reply]
Ok. Fortunately I have a ESP32 devkit too.
Thanks.
